nextest-runner/src/reporter/displayer/progress.rs

```diff
@@ -98,385 +97,107 @@ pub enum ShowProgress {
-                let displayed_count = self
-                    .used_bars
-                    .len()
-                    .saturating_sub(self.overflow_queue.len());
-                let insert_pos = displayed_count;
-                multi.insert(insert_pos, tb.bar.clone());
-                tb.bar.tick(); // Initial render
-                true
-            } else {
-                false
-            }
-        } else {
-            false
-        }
-    }
-
-    fn tick_all(&self) {
-        // XXX tick in sequence rather than random order?
-        for tb in &self.used_bars {
-            tb.bar.tick();
-        }
-    }
-
-    fn finish_and_clear_all(&self) {
-        // XXX finish in sequence rather than random order?
-        for tb in &self.used_bars {
-            tb.bar.finish_and_clear();
-        }
-        for bar in &self.free_bars {
-            bar.finish_and_clear();
-        }
-        if let Some(summary_bar) = &self.summary_bar {
-            summary_bar.bar.finish_and_clear();
+        let max_width = width.saturating_sub(25);
+        let test_width = measure_text_width(&test);
+        if test_width > max_width {
+            test = ansi_get(&test, test_width - max_width, test_width)
         }
+        format!("       {} [{:>9}] {}", status, elapsed, test)
     }
 }
 
-#[derive(Debug)]
-pub(super) struct TestProgressBar {
-    binary_id: RustBinaryId,
-    test_name: String,
-    bar: ProgressBar,
+pub fn measure_text_width(s: &str) -> usize {
+    AnsiCodeIterator::new(s)
+        .filter_map(|(s, is_ansi)| match is_ansi {
+            false => Some(s.chars().count()),
+            true => None,
+        })
+        .sum()
 }
 
-impl IdHashItem for TestProgressBar {
-    type Key<'a> = TestInstanceId<'a>;
+pub fn ansi_get(text: &str, start: usize, end: usize) -> String {
+    let mut pos = 0;
+    let mut res = String::new();
+    for (s, is_ansi) in AnsiCodeIterator::new(text) {
+        if is_ansi {
+            res.push_str(s);
+            continue;
+        } else if pos >= end {
+            continue;
+        }
 
-    fn key(&self) -> Self::Key<'_> {
-        TestInstanceId {
-            binary_id: &self.binary_id,
-            test_name: &self.test_name,
+        for c in s.chars() {
+            let c_width = 1;
+            if start <= pos && pos + c_width <= end {
+                res.push(c);
+            }
+            pos += c_width;
+            if pos > end {
+                // no need to iterate over the rest of s
+                break;
+            }
         }
     }
-
-    id_upcast!();
+    res
 }
 
 #[derive(Debug)]
 pub(super) struct ProgressBarState {
-    multi_progress: MultiProgress,
     bar: ProgressBar,
-    test_bars: Option<TestProgressBars>,
+    stats: RunStats,
+    styles: Styles,
+    running: usize,
+    max_progress_running: MaxProgressRunning,
+    max_running_displayed: usize,
+    running_tests: Option<Vec<RunningTest>>,
```

Comment:
I'd put in a comment here explaining that `None` means that running tests aren't being displayed.